applications/RomApplication/custom_utilities/rom_residuals_utility.h

```diff
@@ -138,17 +140,183 @@ namespace Kratos
+                    }
+
+                }
+
+                #pragma omp for nowait
+                for (int k = 0; k < nconditions;  k++){
+                    ModelPart::ConditionsContainerType::iterator it = cond_begin + k;
+                    //detect if the condition is active or not. If the user did not make any choice the condition is active by default
+                    bool condition_is_active = true;
+                    if ((it)->IsDefined(ACTIVE))
+                        condition_is_active = (it)->Is(ACTIVE);
+                    if (condition_is_active){
+                        Condition::DofsVectorType dofs;
+                        it->GetDofList(dofs, CurrentProcessInfo);
+                        //calculate elemental contribution
+                        mpScheme->CalculateSystemContributions(*it, LHS_Contribution, RHS_Contribution, EquationId, CurrentProcessInfo);
+                        //assemble the elemental contribution - here is where the ROM acts
+                        //compute the elemental reduction matrix PhiElemental
+                        const auto& geom = it->GetGeometry();
+                        if(PsiElemental.size1() != dofs.size() || PsiElemental.size2() != mPetrovGalerkinRomDofs)
+                            PsiElemental.resize(dofs.size(), mPetrovGalerkinRomDofs,false);
+                        RomAuxiliaryUtilities::GetPsiElemental(PsiElemental, dofs, geom, MapPhi);
+                        noalias(row(MatrixResiduals, k+nelements)) = prod(trans(PsiElemental), RHS_Contribution); // The size of the residual will vary only when using more ROM modes, one row per condition
+                    }
+                }
+            }
+        return MatrixResiduals;
+        }
+
+        Matrix GetProjectedGlobalLHS()
+        {
+            const int nElements = static_cast<int>(mpModelPart.Elements().size());
+            const int nConditions = static_cast<int>(mpModelPart.Conditions().size());
+            const auto& nNodes = mpModelPart.NumberOfNodes();
+
+            const auto& r_current_process_info = mpModelPart.GetProcessInfo();
+            
+            const int systemSize = nNodes*mNodalDofs;
+
+            const auto el_begin = mpModelPart.ElementsBegin();
+            const auto cond_begin = mpModelPart.ConditionsBegin();
+
+            //contributions to the system
+            Matrix LHS_Contribution = ZeroMatrix(0,0);
+
+            //vector containing the localization in the system of the different terms
+            Element::EquationIdVectorType EquationId;
+            Matrix APhi = ZeroMatrix(systemSize, mRomDofs);
+
+            #pragma omp parallel firstprivate(nElements, nConditions, LHS_Contribution, EquationId, el_begin, cond_begin)
+            {
+                Matrix PhiElemental;
+                Matrix tempAPhi = ZeroMatrix(systemSize,mRomDofs);
+                Matrix aux;
```

Comment:
Can't you allocate this outside the loop? Also note the style guide naming convention for local variables.

Reply:
Yes, now it is allocated outside.